datafusion/optimizer/src/simplify_predicates.rs

```diff
@@ -0,0 +1,194 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
A minor code reorganization comment: I recommend we put this module into `simplify_expressions` to make it easier to find (`datafusion/optimizer/src/simplify_expressions/predicates.rs`) for example

I view this as a very nice first step for more sophisticated predicate simplification